File: c++/leetcode/804. Unique Morse Code Words.cpp
class Solution
{
public:
  int uniqueMorseRepresentations(vector<string> &words)
  {
    vector<string> morse{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};

    unordered_set<string> seen;

    for (auto str : words)
    {
      string st = "";
      for (auto i : str)
      {
        st += morse[i - 'a'];
      }
      seen.insert(st);
    }

    return seen.size();
  }
};

File: javascript/leetcode/804. Unique Morse Code Words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function (words) {
  const morse = [
    ".-",
    "-...",
    "-.-.",
    "-..",
    ".",
    "..-.",
    "--.",
    "....",
    "..",
    ".---",
    "-.-",
    ".-..",
    "--",
    "-.",
    "---",
    ".--.",
    "--.-",
    ".-.",
    "...",
    "-",
    "..-",
    "...-",
    ".--",
    "-..-",
    "-.--",
    "--..",
  ];

  const seen = new Set();

  for (let str of words) {
    let st = "";
    for (let i of str) {
      const code = i.charCodeAt(0) - 97;
      st += morse[code];
    }
    seen.add(st);
  }

  return seen.size;
};
